src/main/java/io/quarkus/search/app/quarkusio/QuarkusIO.java

```diff
@@ -364,47 +476,58 @@ private static Set<String> combine(Set<String> a, Set<String> b) {
         if (parsedUrl.startsWith("http")) {
             // we are looking at a quarkiverse guide:
             return createQuarkiverseGuide(quarkusVersion, type, parsedGuide, summaryKey);
         } else {
-            return createCoreGuide(urlBase, quarkusVersion, type, parsedGuide, summaryKey);
+            return createCoreGuide(cloneDirectory, quarkusVersion, type, parsedGuide, summaryKey, language, messages);
         }
     }
 
-    private Guide createCoreGuide(URI urlBase, String quarkusVersion, String type, Map<String, Object> parsedGuide,
-            String summaryKey) {
+    private Guide createCoreGuide(GitCloneDirectory cloneDirectory, String quarkusVersion, String type,
+            Map<String, Object> parsedGuide,
+            String summaryKey, Language language, Catalog messages) {
         Guide guide = new Guide();
         guide.quarkusVersion = quarkusVersion;
-        guide.language = Language.ENGLISH;
+        guide.language = language;
         guide.origin = toString(parsedGuide.get("origin"));
         if (guide.origin == null) {
             guide.origin = QUARKUS_ORIGIN;
         }
         guide.type = type;
-        guide.title.set(Language.ENGLISH, renderMarkdown(toString(parsedGuide.get("title"))));
-        guide.summary.set(Language.ENGLISH, renderMarkdown(toString(parsedGuide.get(summaryKey))));
+        guide.title.set(language, renderMarkdown(translate(messages, toString(parsedGuide.get("title")))));
+        guide.summary.set(language, renderMarkdown(translate(messages, toString(parsedGuide.get(summaryKey)))));
         String parsedUrl = toString(parsedGuide.get("url"));
-        guide.url = httpUrl(urlBase, quarkusVersion, parsedUrl);
-        guide.htmlFullContentProvider.set(Language.ENGLISH,
-                new GitInputProvider(mainRepository.git(), mainRepository.pagesTree(),
-                        htmlPath(quarkusVersion, parsedUrl)));
+        guide.url = httpUrl(localizedSiteUris.get(language), quarkusVersion, parsedUrl);
+        GitInputProvider gitInputProvider = new GitInputProvider(cloneDirectory.git(), cloneDirectory.pagesTree(),
+                htmlPath(language, quarkusVersion, parsedUrl));
+        guide.htmlFullContentProvider.set(language, gitInputProvider);
+
+        if (!gitInputProvider.isFileAvailable()) {
+            // if  a file is not present we do not want to add such guide. Since if the html is not there
+            // it means that users won't be able to open it on the site, and returning it in the search results make it pointless.
+
+            // Since this file was listed in the yaml of a rev from which the site was built the html should've been present,
+            // but is missing for some reason that needs investigation:
+            failureCollector.warning(
+                    FailureCollector.Stage.TRANSLATION,
+                    "Guide " + guide + " is ignored since we were not able to find an HTML content file for it.");
+            return null;
```

Comment:
Oddly I'm still getting warnings with your changes:

* Lots of them about `es` + 3.6; those are probably related to the problem @gsmet fixed earlier this week, so they'll go away on the next sync (hopefully)
* One is more suspicious: https://es.quarkus.io/version/main/guides/config-secrets. Something's up, no?

Reply:
The submodule points to this rev of the main repository, where the guide is available:

https://github.com/quarkusio/quarkusio.github.io/blob/11e73734dc95cc20e0f91a4c7881a98f87f08528/_data/versioned/main/index/quarkus.yaml#L2098

But for some reason, it is missing in the generated-translated guides here:
https://github.com/quarkusio/es.quarkus.io/tree/docs/docs/version/main/guides

hopefully this gets refreshed with the next sync 🤞🏻🙈

Reply:
Ok, so a valid warning, works for me. Thanks for having a look.